src/utils/media-devices-manager.js

```diff
@@ -127,10 +133,10 @@ export default class MediaDevicesManager extends EventEmitter {
       if (this._permissionsStatus[MediaDevices.MICROPHONE] === PermissionStatus.DENIED) {
         return NO_DEVICE_ID;
       } else {
-        return this.deviceIdForMicDeviceLabel(this.selectedMicLabel) || DEFAULT_DEVICE_ID;
+        return this.deviceIdForMicDeviceLabel(this.selectedMicLabel) || this.defaultInputDeviceId;
```

Comment:
Again its pretty hard to trace through all the derived values. but I think `deviceIdForMicDeviceLabel` ends up calling the `micDevices` getter which ends up checking the same conditions that the surrounding if statement here is checking. That means it would already be returning `[{ value: NO_DEVICE_ID, label: "None" }]` as `micDevices`, which would then not pass the filter in `deviceIdForMicDeviceLabel`, so that would return `undefined`. I think this means you can just drop the entire if statement and just keep this line, but it should be `|| NO_DEVICE_ID` instead, which is I think is more correct anyway.